project_template/src/operations.c

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#include "operations.h"
+int32_t az_exampleshortname_add_two_numbers(int32_t a, int32_t b) {

```

Comment:
>{ [](start = 66, length = 1)

Are we enforcing the curly bracket style? If yes, IMO, we should go with Allman style (curly braces in new line for each block) in our samples. This is too Java-ish.

Reply:
I have added a clang-format file to the root (and reformatted source files). It's based on the LLVM style.

Reply:
seconded

---
In reply to: [309417312](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309417312) [](ancestors = 309417312)

Reply:
Any motivations/arguments for choosing this specific style? 

---
In reply to: [309447254](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309447254) [](ancestors = 309447254)

Reply:
LLVM is what we use in the STL, and we've discussed this in meetings with the azure-native team and they seem to like it. I don't really care as long as everyone agrees and there's a clang-format file.

https://clang.llvm.org/docs/ClangFormatStyleOptions.html has documentations on the various styles and clang-format options.

Reply:
there are parts of LLVM that seem to clash with several different code bases we have. 
First LLVM uses camel case in which case it clashes with this file. Our team prefers snake case like you use here since that also aligns with threadx and their general practices. We also use Allman style braces as does threadx.

---
In reply to: [309470098](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309470098) [](ancestors = 309470098)

Reply:
The llvm style (and clang format in general) will never, ever, ever rename a type. 

We (the vclibs and the azure-native teams) are not trying to import and follow the llvm style guide, we were just OK with using their formatting style as a base.